utils/src/main/kotlin/OrtAuthenticator.kt

```diff
@@ -74,15 +74,54 @@ class OrtAuthenticator(private val original: Authenticator? = null) : Authentica
     }
 
     override fun getPasswordAuthentication(): PasswordAuthentication? {
-        if (requestorType == RequestorType.PROXY) {
-            val proxySelector = ProxySelector.getDefault()
-            if (proxySelector is OrtProxySelector) {
-                val type = requestingProtocol.toProxyType() ?: return null
-                val proxy = Proxy(type, InetSocketAddress(requestingHost, requestingPort))
-                return proxySelector.getProxyAuthentication(proxy)
+        when (requestorType) {
+            RequestorType.PROXY -> {
+                val proxySelector = ProxySelector.getDefault()
+                if (proxySelector is OrtProxySelector) {
+                    val type = requestingProtocol.toProxyType() ?: return null
+                    val proxy = Proxy(type, InetSocketAddress(requestingHost, requestingPort))
+                    return proxySelector.getProxyAuthentication(proxy)
+                }
+            }
+
+            RequestorType.SERVER -> {
+                val netrcFile = getUserHomeDirectory().resolve(".netrc")
+                if (netrcFile.isFile) {
+                    getNetrcAuthentication(netrcFile.readText(), requestingHost)?.let { return it }
+                }
+
+                // TODO: Add support for .authinfo files (which use the same syntax as .netrc files) once Git.kt does
+                //       not call the Git CLI anymore which only suuports .netrc files.
             }
+
+            null -> log.warn { "No requestor type set for password authentication." }
         }
 
         return super.getPasswordAuthentication()
     }
 }
+
+fun getNetrcAuthentication(contents: String, machine: String): PasswordAuthentication? {
+    val lines = contents.lines().mapNotNull { line ->
```

Comment:
What do you think about adding the URL to the netrc file format here as a code comment next to the parsing?

Reply:
Done.